config/config.go

```diff
@@ -1014,6 +1015,12 @@ func (c *RelabelConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if (c.Action == RelabelReplace || c.Action == RelabelHashMod) && c.TargetLabel == "" {
 		return fmt.Errorf("relabel configuration for %s action requires 'target_label' value", c.Action)
 	}
+	if c.Action == RelabelReplace && !relabelTarget.Match([]byte(c.TargetLabel)) {
+		return fmt.Errorf("%q is invalid 'target_label' for %s action", c.TargetLabel, c.Action)
+	}
+	if c.Action == RelabelHashMod && !model.LabelNameRE.Match([]byte(c.TargetLabel)) {
```

Comment:
The relabelTarget regex makes sure that the label starts with a-zA-Z_ and containing a-zA-Z_0-9 after that while allowing regexp interpolations at any position. If its RelabelHashMod the old LabelNameRE is used as no interpolation is supported for that action.